pycaret/time_series/forecasting/oop.py

```diff
@@ -1700,6 +1704,11 @@ def create_model(
         is set to False.
 
         """
+
+        if not self._setup_ran:
```

Comment:
We should use a decorator first and foremost.

Reply:
Seems an overkill and less flexible - don't you think. For example, what if I want to check a few things and only conditionally issue this message (such as in `predict_model`. Having a private method for raising this error would provide more flexibility.

Reply:
It would be less flexible, but for the PR as it is right now, a decorator would fit the best. But I am fine with a private method also. Whatever we do, however, we should somehow reuse the code.

Reply:
O, we went with a private method for now. Currently, it is in the time series module only. Later we may add it to the base class itself.